src/releaseNotesProvider.ts

```diff
@@ -0,0 +1,180 @@
+import { env, ExtensionContext } from "vscode";
+import * as vscode from "vscode";
+import * as semver from "semver";
+import { Remarkable } from "remarkable";
+import http from "https";
+
+const versionKey = "metals-server-version";
+
+// prettier-ignore
+const releaseNotesMarkdownUrl: Record<string, string> = {
+  "0.11.5": "https://raw.githubusercontent.com/scalameta/metals/main/website/blog/2022-04-28-aluminium.md",
+  "0.11.6": "https://raw.githubusercontent.com/scalameta/metals/main/website/blog/2022-06-03-aluminium.md",
+};
+
+/**
+ * Show release notes if possible, swallow errors since its not a crucial feature.
+ * Treats snapshot versions like 0.11.6+67-926ec9a3-SNAPSHOT as a 0.11.6.
+ */
+export async function showReleaseNotesIfNeeded(
+  context: ExtensionContext,
+  serverVersion: string,
+  outputChannel: vscode.OutputChannel
+) {
+  try {
+    await showReleaseNotes(context, serverVersion);
+  } catch (error) {
+    outputChannel.appendLine(
+      `Couldn't show release notes for Metals ${serverVersion}`
+    );
+    outputChannel.appendLine(`${error}`);
+  }
+}
+
+async function showReleaseNotes(
+  context: ExtensionContext,
+  currentVersion: string
+): Promise<void> {
+  const state = context.globalState;
+
+  const version = getVersion();
+  const releaseNotesUrl = version
+    ? releaseNotesMarkdownUrl[version]
+    : undefined;
+  if (isNotRemote() && version && releaseNotesUrl) {
+    await showPanel(version, releaseNotesUrl);
+  }
+
+  async function showPanel(version: string, releaseNotesUrl: string) {
+    const releaseNotes = await getReleaseNotesMarkdown(releaseNotesUrl);
+
+    if (typeof releaseNotes === "string") {
+      const panel = vscode.window.createWebviewPanel(
+        `scalameta.metals.whatsNew`,
+        `Metals ${version} release notes`,
+        vscode.ViewColumn.One
+      );
+
+      panel.webview.html = releaseNotes;
+      panel.reveal();
+
+      // set wrong value for local development
+      // const newVersion = "0.11.5";
+
+      const newVersion = version;
+
+      // Update current device's latest server version when there's no value or it was a older one.
+      // Then sync this value across other devices.
+      state.update(versionKey, newVersion);
+      state.setKeysForSync([versionKey]);
+
+      context.subscriptions.push(panel);
+    }
+  }
+
+  /**
+   * Don't show panel for remote environment because it installs extension on every time.
+   * TODO: what about wsl?
```

Comment:
What if we instead only show the release notes if the version was set previously. If the version was not set then we ignore it, but we could start from the next release?

So for now empty means user used the previous version. After the next release we treat empty as new installers, WSL etc. where we don't need to show anything (or we could show a more global welcome view)

Reply:
I'm afraid that I don't understand 😅

Reply:
I was just thinking that maybe the first time user install Metals and no version was set up, we would run only the welcome view. (or nothing at all). This is however a problem for when we want to show the release notes first time for even existing users. So we could at first assume that no one is the first user and always show release notes if no metals version was saved globally. And after the next release we can safely revert the bahaviour to only show release notes if not server version was specified previously.

This is however just a vague idea, so not a blocker at all for me.

Reply:
> I was just thinking that maybe the first time user install Metals and no version was set up, we would run only the welcome view.

For welcome view, if I understand correctly, there is [special contribution point](https://code.visualstudio.com/api/references/contribution-points#contributes.walkthroughs) so I think we can show both welcome and release notes.

IMO there is no harm in showing release notes for new people, this might even encourage them to visit scalemta.org/metals so if you're OK, like you said, I'll leave this as it is.

Reply:
Ok sure, let's see if it causes any issues for people (it shouldn't really)